deepsubtree.go

```diff
@@ -186,38 +113,56 @@ func (dst *DeepSubTree) linkNode(node *Node) error {
 	if len(node.leftHash) > 0 {
 		if node.leftNode == nil {
 			node.leftNode, _ = dst.ndb.GetNode(node.leftHash)
-		} else if !bytes.Equal(node.leftNode.hash, node.leftHash) {
-			return fmt.Errorf(
-				"for node: %s, leftNode hash: %s and node leftHash: %s do not match",
-				node.hash,
-				node.leftNode.hash,
-				node.leftHash,
-			)
 		}
 	}
 	if len(node.rightHash) > 0 {
 		if node.rightNode == nil {
 			node.rightNode, _ = dst.ndb.GetNode(node.rightHash)
-		} else if !bytes.Equal(node.rightNode.hash, node.rightHash) {
+		}
+	}
+	return nil
+}
+
+func (dst *DeepSubTree) verifyOperation(operation Operation, key []byte, value []byte) error {
+	// Verify operation is on top, look at the witness data and add the relevant existence proofs
+	if dst.witnessData != nil && dst.operationCounter < len(dst.witnessData) {
```

Comment:
Shouldn't you return error is `dst.operationCounter >= len(dst.witnessData)` (`else` for this `if`).

Reply:
Yes, thanks for pointing this out. Addressed.